File: DGVTemplateEditor/Types/Design.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;

namespace DGVTemplateEditor.Types
{
	public class Design
	{
		public Design()
		{
		}

		public Design(DGVTemplate T)
		{
			Columns = T.Columns.Select(c => new DesignColumn(c)).ToList();
		}

		public List<DesignColumn> Columns { get; set; }

		[Description("Ширина столбца")]
		public Font Font { get; set; }

		public static Design FromTemplate(DGVTemplate T)
		{
			return new Design(T);
		}

		public DGVTemplate ToTemplate()
		{
			return new DGVTemplate
			{
				Columns = Columns.Select(c => c.ToTemplate()).ToList()
			};
		}
	}
}

File: JANL.VB/Types/Design.vb
Namespace Types

	Friend Class Design

		Public Sub New()
		End Sub

		Public Sub New(T As DGVTemplate)
			Columns = T.Columns.Select(Function(c) New DesignColumn(c)).ToList()
		End Sub

		Public Columns As List(Of DesignColumn)

		'<Description("Ширина столбца")>
		'Public Font Font { Get; Set; }

		Public Shared Function FromTemplate(T As DGVTemplate) As Design
			Return New Design(T)
		End Function

		Public Function ToTemplate() As DGVTemplate
			Return New DGVTemplate With {.Columns = Columns.Select(Function(c) c.ToTemplate()).ToList()}
		End Function

	End Class
End Namespace